realm/realm-annotations-processor/src/main/java/io/realm/processor/ClassMetaData.java

```diff
@@ -333,7 +333,14 @@ public boolean isNullable(VariableElement variableElement) {
         // primary keys cannot be nullable
         if (hasPrimaryKey()) {
             if (variableElement.equals(getPrimaryKey())) {
-                return false;
+                // Check if element type is String
+                String elementTypeCanonicalName = variableElement.asType().toString();
+                String columnType = Constants.JAVA_TO_COLUMN_TYPES.get(elementTypeCanonicalName);
+                if (columnType != null && columnType.equals("RealmFieldType.STRING")) {
```

Comment:
Why is only Strings keys allowed to be Null, it seems a bit arbitrary to me? We have nullable integers and booleans?